boa_engine/src/builtins/function/mod.rs

```diff
@@ -275,23 +277,152 @@ pub struct BuiltInFunctionObject;
+    ///
+    /// [spec]: https://tc39.es/ecma262/#sec-function-p1-p2-pn-body
+    /// [mdn]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Function/Function
     fn constructor(
         new_target: &JsValue,
-        _: &[JsValue],
+        args: &[JsValue],
         context: &mut Context,
     ) -> JsResult<JsValue> {
+        Self::create_dynamic_function(new_target, args, context).map(Into::into)
+    }
+
+    /// `CreateDynamicFunction ( constructor, newTarget, kind, args )`
+    ///
+    /// More information:
+    ///  - [ECMAScript reference][spec]
+    ///
+    /// [spec]: https://tc39.es/ecma262/#sec-createdynamicfunction
+    fn create_dynamic_function(
+        new_target: &JsValue,
+        args: &[JsValue],
+        context: &mut Context,
+    ) -> JsResult<JsObject> {
         let prototype =
             get_prototype_from_constructor(new_target, StandardConstructors::function, context)?;
+        if let Some((body_arg, args)) = args.split_last() {
+            let parameters =
+                if args.is_empty() {
+                    FormalParameterList::empty()
+                } else {
+                    let mut parameters = Vec::with_capacity(args.len());
+                    for arg in args {
+                        parameters.push(arg.to_string(context)?);
+                    }
+                    let mut parameters = parameters.join(",");
+                    parameters.push(')');
+
+                    let parameters = match Parser::new(parameters.as_bytes())
+                        .parse_formal_parameters(context.interner_mut(), false, false)
+                    {
+                        Ok(parameters) => parameters,
+                        Err(e) => {
+                            return context.throw_syntax_error(format!(
+                                "failed to parse function parameters: {e}"
+                            ))
+                        }
+                    };
+                    parameters
+                };
+
+            let body_arg = body_arg.to_string(context)?;
+
+            let body = match Parser::new(body_arg.as_bytes()).parse_function_body(
+                context.interner_mut(),
+                false,
+                false,
+            ) {
+                Ok(statement_list) => statement_list,
+                Err(e) => {
+                    return context
+                        .throw_syntax_error(format!("failed to parse function body: {e}"))
+                }
+            };
+
+            // Early Error: If BindingIdentifier is present and the source text matched by BindingIdentifier is strict mode code,
+            // it is a Syntax Error if the StringValue of BindingIdentifier is "eval" or "arguments".
+            if body.strict() {
+                for parameter in parameters.parameters.iter() {
+                    for name in parameter.names() {
+                        if [Sym::EVAL, Sym::ARGUMENTS].contains(&name) {
```

Comment:
Since we are using symbols here, and integer comparisons are very cheap, I think it will be faster to just have two equals in an "or" comparison.